snap/container.go

```diff
@@ -387,18 +404,25 @@ func validateContainer(c Container, needsrx, needsx, needsr, needsf, noskipd map
 		return err
 	}
 	if len(seen) != len(needsx)+len(needsrx)+len(needsr) {
+		var firstPath string
 		for _, needs := range []map[string]bool{needsx, needsrx, needsr} {
 			for path := range needs {
 				if !seen[path] {
 					logf("in %s %q: path %q does not exist", contType, name, path)
+					if firstPath == "" {
+						firstPath = path
+					}
 				}
 			}
 		}
-		return ErrMissingPaths
+		// TODO: aggregate path errors not just the first one
+		return fmt.Errorf("%w: path %q does not exist", ErrMissingPaths, firstPath)
 	}
 
-	if hasBadModes {
-		return ErrBadModes
+	if firstBadModeErr != nil {
+		// TODO: fmt.Errorf("%w: %w", ErrBadModes, firstBadModeErr) when using go 1.20+
+		// TODO: aggregate bad mode errors not just the first one
+		return fmt.Errorf("%w: %v", ErrBadModes, firstBadModeErr)
```

Comment:
Can we use %w twice? I think we can but I'm not sure if this goes all the way back to go 1.18.

Reply:
Unfortunately we can't, I  believe we need go 1.20 for that https://tip.golang.org/doc/go1.20#errors